legacy.go

```diff
@@ -667,6 +669,8 @@ func (w *legacyLayerWriter) AddLink(name string, target string) error {
 	var selectedRoot string
 	if _, ok := w.addedFiles[target]; ok {
 		selectedRoot = w.destRoot
+	} else if _, ok := w.utilityVMPendingLinks[relativePathToKey(target)]; ok {
```

Comment:
If you set `w.addedFiles[x]` to true whenever you set `w.utilityVMPendingLinks[rptk(x)]`, then this extra condition goes away. Of course then you'll have to clear it in `Remove`.

I wonder if you can merge `addedFiles`, `pendingLinks`, and `utilityVMPendingLinks` together in some way as a simplification.